lenskit-core/src/main/java/org/grouplens/lenskit/data/event/LikeBatch.java

```diff
@@ -20,18 +20,102 @@
  */
 package org.grouplens.lenskit.data.event;
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
 /**
  * Generic interface representing a batch of unary interactions.  Events of this type will generally
  * not have timestamps.  An example of a use of this event type is to represent play counts in a
  * summarized data set.
+ * <p>
+ * Like batches can be constructed with {@link LikeBatchBuilder} or {@link #create(long, long, int)}.
  *
  * @since 2.2
  * @see Like
  * @see Events#likeBatch(long, long, int)
  */
-public interface LikeBatch extends Event {
+public class LikeBatch implements Event {
+    private final long userId;
+    private final long itemId;
+    private final int count;
+
+    LikeBatch(long uid, long iid, int ct) {
+        userId = uid;
+        itemId = iid;
+        count = ct;
+    }
+
+    public static LikeBatchBuilder newBuilder() {
+        return new LikeBatchBuilder();
+    }
+
+    public static LikeBatch create(long user, long item, int count) {
+        return new LikeBatch(user, item, count);
+    }
+
+    @Override
+    public long getUserId() {
+        return userId;
+    }
+
+    @Override
+    public long getItemId() {
+        return itemId;
+    }
+
     /**
-     * Get the number of likes
+     * Get the number of items liked.
+     * @return The number of items liked.
      */
-    int getCount();
+    public int getCount() {
```

Comment:
Next function - a good reason to have getters (interface compatibility) not a good reason for final variables to be private.